lib/ansible/plugins/lookup/varnames.py

```diff
@@ -0,0 +1,79 @@
+# (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    lookup: varnames
+    author: Ansible Core
+    version_added: "2.8"
+    short_description: Lookup matching variable names
+    description:
+      - Retrieves a list of matching Ansible variable names.
+    options:
+      _terms:
+        description: List of patterns to match to variable names.
+        required: True
+"""
+
+EXAMPLES = """
+- name: List variables that start with qz_
+  debug: msg="{{ lookup('varnames', 'qz_.+')}}"
```

Comment:
Shouldn't this be
```suggestion
  debug: msg="{{ lookup('varnames', '^qz_.+')}}"
```
? Otherwise, it should also find a variable called `pqz_1`.